docker-build.sh

```diff
@@ -0,0 +1,355 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+##############################################################
+# Docker-based build script for StarRocks
+# This script uses the official StarRocks dev-env Docker image
+# to build the project in a consistent Ubuntu environment.
+#
+# Usage:
+#    ./docker-build.sh --help
+# Examples:
+#    ./docker-build.sh                                    # build all (FE + BE)
+#    ./docker-build.sh --fe                               # build Frontend only
+#    ./docker-build.sh --be                               # build Backend only
+#    ./docker-build.sh --fe --clean                       # clean and build Frontend
+#    ./docker-build.sh --fe --be --clean                  # clean and build both
+#    ./docker-build.sh --shell                            # open interactive shell
+#    ./docker-build.sh --test                             # run tests
+##############################################################
+
+set -euo pipefail
+
+# Configuration
+DOCKER_IMAGE="${STARROCKS_DEV_ENV_IMAGE:-starrocks/dev-env-ubuntu:latest}"
+CONTAINER_NAME="starrocks-build-$(date +%s)"
```

Comment:
include uid/username as part of the container-name, in case a multi-user shared dev machine, the container_name won't be conflicted with each other.